lib/logster/logger.rb

```diff
@@ -15,18 +15,21 @@ def initialize(store)
       @override_levels = nil
       @chained = []
       @skip_store = false
+      @mutex = Mutex.new
     end
 
     def override_level=(val)
       tid = Thread.current.object_id
 
-      ol = @override_levels
-      if val.nil? && ol && ol.key?(tid)
-        ol.delete(tid)
-        @override_levels = nil if ol.length == 0
-      elsif val
-        (@override_levels ||= {})[tid] = val
-      end
+      @mutex.synchronize {
+        ol = @override_levels
+        if val.nil? && ol && ol.key?(tid)
+          ol.delete(tid)
+          @override_levels = nil if ol.length == 0
+        elsif val
+          (@override_levels ||= {})[tid] = val
+        end
+      }
```

Comment:
This only avoids some the issues, because the read side (in `level`) is not synchronized.
So if `override_level=` is called concurrently to `level`, then you still have concurrent hash accesses.

I think using thread/fiber locals would be a much safer and still fast way to implement this correctly